crates/types/src/blockchain/header.rs

```diff
@@ -37,6 +47,58 @@ pub struct BlockHeader {
     metadata: Option<BlockHeaderMetadata>,
 }
 
+impl From<BlockHeaderV1> for BlockHeader {
+    fn from(v1: BlockHeaderV1) -> Self {
+        BlockHeader::V1(v1)
+    }
+}
+
+impl BlockHeader {
+    /// Getter for consensus portion of header
+    pub fn consensus(&self) -> &ConsensusHeader<GeneratedConsensusFields> {
```

Comment:
Hmm, the `BlockHeader` is mostly a composition of the consensus and application headers, and, likely, updating one of them will break these getters=)

Maybe we want to consider add getter for some specific fields like `applciation_hash`, `transactions_root`, etc.

In that case, maybe instead of providing a mutable reference getter like `applciation_hash_mut`, we want to provide an explicit setter(`set_applciation_hash`) instead.

Reply:
Yep. I agree. I'll do that.

Reply:
Hmm. Some of these might be better off under a test feature. On the other hand, I'm not exactly sure what the risk of having these mutable though.